### PR TITLE
Update errors to use go1.13 semantics

### PR DESCRIPTION
**Pull request by mtibben**
Remove [github.com/pkg/errors](https://github.com/pkg/errors) and use [go 1.13 errors](https://go.dev/blog/go1.13-errors) instead

I have:
 - [ ] Added tests covering the bug / feature (see [testing](https://github.com/99designs/gqlgen/blob/master/TESTING.md))
 - [ ] Updated any relevant documentation (see [docs](https://github.com/99designs/gqlgen/tree/master/docs/content))

### CODE REVIEW

File: cmd/gen.go [hunk 1]
```diff
@@ -26,7 +26,7 @@ var genCmd = &cli.Command{
 			}
 		} else {
 			cfg, err = config.LoadConfigFromDefaultLocations()
-			if os.IsNotExist(errors.Cause(err)) {
+			if os.IsNotExist(errors.Unwrap(err)) {
```
Comment (lwc) [inline cmd/gen.go:29]: I think this simplifies to:
```suggestion
			if errors.Is(err, fs.ErrNotExist) {
```

File: codegen/config/config_test.go [hunk 2]
```diff
@@ -105,7 +105,7 @@ func TestLoadDefaultConfig(t *testing.T) {
 		require.NoError(t, err)
 
 		cfg, err = LoadDefaultConfig()
-		require.True(t, os.IsNotExist(errors.Cause(err)))
+		require.True(t, os.IsNotExist(errors.Unwrap(err)))
```
Comment (lwc) [inline codegen/config/config_test.go:108]:
```suggestion
		require.True(t, errors.Is(err, fs.ErrNotExist))
```

File: plugin/resolvergen/resolver.go [hunk 3]
```diff
@@ -144,7 +144,7 @@ func (m *Plugin) generatePerSchema(data *codegen.Data) error {
 		}
 	}
 
-	if _, err := os.Stat(data.Config.Resolver.Filename); os.IsNotExist(errors.Cause(err)) {
+	if _, err := os.Stat(data.Config.Resolver.Filename); os.IsNotExist(errors.Unwrap(err)) {
```
Comment (lwc) [inline plugin/resolvergen/resolver.go:147]:
```suggestion
	if _, err := os.Stat(data.Config.Resolver.Filename); errors.Is(err, fs.ErrNotExist) {
```

File: plugin/servergen/server.go [hunk 4]
```diff
@@ -29,7 +29,7 @@ func (m *Plugin) GenerateCode(data *codegen.Data) error {
 		ResolverPackageName: data.Config.Resolver.ImportPath(),
 	}
 
-	if _, err := os.Stat(m.filename); os.IsNotExist(errors.Cause(err)) {
+	if _, err := os.Stat(m.filename); os.IsNotExist(errors.Unwrap(err)) {
```
Comment (lwc) [inline plugin/servergen/server.go:32]:
```suggestion
	if _, err := os.Stat(m.filename); errors.Is(err, fs.ErrNotExist) {
```

**Comment by wilhelmeek** [comment]
Nice one!

**Comment by mtibben** [comment]
@lwc your proposed changes require the latest go and cause some cascading changes to I've created a new branch for that https://github.com/99designs/gqlgen/compare/go-1.17

**Pull request merged**